packages/cli/create-strapi-app/src/index.ts

```diff
@@ -0,0 +1,243 @@
+  .option('--quickstart', 'Quickstart app creation (deprecated)')
+  .option('--no-run', 'Do not start the application after it is created.')
+
+  // setup options
+  .option('--ts, --typescript', 'Initialize the project with TypeScript (default)')
+  .option('--js, --javascript', 'Initialize the project with Javascript')
+
+  // Package manager options
+  .option('--use-npm', 'Use npm as the project package manager')
+  .option('--use-yarn', 'Use yarn as the project package manager')
+  .option('--use-pnpm', 'Use pnpm as the project package manager')
+
+  // dependencies options
+  .option('--install', 'Install dependencies')
+  .option('--no-install', 'Do not install dependencies')
+
+  // Cloud options
+  .option('--skip-cloud', 'Skip cloud login and project creation')
+
+  // Example app
+  .option('--example', 'Use an example app')
+  .option('--no-example', 'Do not use an example app')
+
+  // git options
+  .option('--git-init', 'Initialize a git repository')
+  .option('--no-git-init', 'Do no initialize a git repository')
+
+  // Database options
+  .option('--dbclient <dbclient>', 'Database client')
+  .option('--dbhost <dbhost>', 'Database host')
+  .option('--dbport <dbport>', 'Database port')
+  .option('--dbname <dbname>', 'Database name')
+  .option('--dbusername <dbusername>', 'Database username')
+  .option('--dbpassword <dbpassword>', 'Database password')
+  .option('--dbssl <dbssl>', 'Database SSL')
+  .option('--dbfile <dbfile>', 'Database file path for sqlite')
+  .option('--skip-db', 'Skip database configuration')
+
+  .option('--template <template>', 'Specify a Strapi template')
+  .option('--template-branch <templateBranch>', 'Specify a branch for the template')
+  .option('--template-path <templatePath>', 'Specify a path to the template inside the repository')
+
+  .description('create a new application');
+
+async function run(args: string[]): Promise<void> {
+  const options = command.parse(args).opts<Options>();
+  const directory = command.args[0];
+
+  logger.title(
+    'Strapi',
+    `${chalk.green(chalk.bold(`v${version}`))} ${chalk.bold("🚀 Let's create your new project")}\n`
+  );
+
+  if (
+    (options.javascript !== undefined || options.typescript !== undefined) &&
+    options.template !== undefined
+  ) {
+    logger.fatal(
+      `You cannot use ${chalk.bold('--javascript')} or ${chalk.bold('--typescript')} with ${chalk.bold('--template')}`
+    );
+  }
+
+  if (options.javascript === true && options.typescript === true) {
+    logger.fatal(
+      `You cannot use both ${chalk.bold('--typescript')} (--ts) and ${chalk.bold('--javascript')} (--js) flags together`
+    );
+  }
+
+  // Only prompt the example app option if there is no template option
+  if (options.example === true && options.template !== undefined) {
+    logger.fatal(`You cannot use ${chalk.bold('--example')} with ${chalk.bold('--template')}`);
+  }
+
+  if (options.template !== undefined && options.template.startsWith('-')) {
```

Comment:
What's the reason of checking this specific usecase?

Reply:
because commander allows doing this `cli --template --typescript` and will think --typescript is the name of the template instead of an extra option

Reply:
Interesting, make sense then